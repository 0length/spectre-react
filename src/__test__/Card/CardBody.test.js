import React from 'react'
import { shallow } from 'enzyme'

import { CardBody } from '../..'

describe('CardBody', () => {
  it('Should render default .card-body markup', () => {
    const wrapper = shallow(<CardBody />)

    expect(wrapper.type()).toBe('div')
    expect(wrapper.hasClass('card-body')).toBe(true)
  })

  it('Should render children', () => {
    const wrapper = shallow(
      <CardBody>
        <p>hello world</p>
      </CardBody>
    )

    expect(wrapper.children().length).toBe(1)
    expect(wrapper.find('p').length).toBe(1)
  })

  it('Should pass classnames', () => {
    const wrapper = shallow(<CardBody className='extra' />)

    expect(wrapper.hasClass('extra')).toBe(true)
    expect(wrapper.hasClass('card-body')).toBe(true)
  })

  it('Should render as a custom element', () => {
    const wrapper = shallow(<CardBody renderAs='main'>custom</CardBody>)

    expect(wrapper.text()).toBe('custom')
    expect(wrapper.hasClass('card-body')).toBe(true)
    expect(wrapper.type()).toBe('main')
  })

  it('Should match the snapshot', () => {
    const wrapper = shallow(<CardBody />)

    expect(wrapper).toMatchSnapshot()
  })
})
