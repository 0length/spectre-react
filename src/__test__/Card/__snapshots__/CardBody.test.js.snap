// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBody Should match the snapshot 1`] = `
<div
  className="card-body"
/>
`;
